packages/docusaurus-theme-translations/update.js

```diff
@@ -209,14 +236,43 @@ ${logKeys(untranslatedKeys)}`),
 }
 
 async function updateCodeTranslations() {
-  logSection('Will update base file');
-  const {baseFile, localesFiles} = await getCodeTranslationFiles();
-  const baseFileMessages = await updateBaseFile(baseFile);
-
-  // eslint-disable-next-line no-restricted-syntax
-  for (const localeFile of localesFiles) {
-    logSection(`Will update ${path.basename(localeFile)}`);
-    await updateLocaleCodeTranslations(localeFile, baseFileMessages);
+  /* eslint-disable no-continue */
+  for (const theme of Themes) {
+    const {baseFile, localesFiles} = await getCodeTranslationFiles(theme.name);
+    logSection(`Will update base file for ${theme.name}`);
+    const baseFileMessages = await updateBaseFile(baseFile, theme.src);
+    const [, newLocale] = process.argv;
+
+    if (newLocale) {
+      const newLocalePath = getThemeLocalePath(newLocale, theme.name);
+
+      if (!fs.existsSync(newLocalePath)) {
+        await writeMessagesFile(newLocalePath, baseFileMessages);
+        console.error(
+          chalk.green(
+            `Locale file ${path.basename(newLocalePath)} have been created.`,
+          ),
+        );
+      } else {
+        console.error(
+          chalk.red(
+            `Locale file ${path.basename(newLocalePath)} was already created!`,
+          ),
+        );
+      }
+
+      continue;
+    }
+
+    for (const localeFile of localesFiles) {
+      logSection(
+        `Will update ${path.basename(
+          path.dirname(localeFile),
+        )} locale in ${path.basename(localeFile, path.extname(localeFile))}`,
+      );
+
+      await updateLocaleCodeTranslations(localeFile, baseFileMessages);
+    }
```

Comment:
what do you think about using else instead?

```suggestion
    } else {
      for (const localeFile of localesFiles) {
        logSection(
          `Will update ${path.basename(
            path.dirname(localeFile),
          )} locale in ${path.basename(localeFile, path.extname(localeFile))}`,
        );

        await updateLocaleCodeTranslations(localeFile, baseFileMessages);
      }
    }
```

Reply:
That's definitely better, thanks!